maven-core/src/main/java/org/apache/maven/plugin/MavenPluginPrerequisiteChecker.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.maven.plugin;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.function.Consumer;
+
+import org.apache.maven.plugin.descriptor.PluginDescriptor;
+
+/**
+ * Service responsible for checking if plugin's prerequisites are met.
+ */
+@FunctionalInterface
+public interface MavenPluginPrerequisiteChecker extends Consumer<PluginDescriptor>
+{
+    /**
+     * 
+     * @param pluginDescriptor
+     * @throws IllegalStateException in case the checked prerequisite is not met
+     */
+    void accept( PluginDescriptor pluginDescriptor ) throws IllegalStateException;
```

Comment:
throws is redundant for runtime exceptions

Reply:
right, but I would still list it explicitly here for clarity reasons that implementations are expected to throw this.

Reply:
Not necessary, we have the Javadoc. See Effective Java by J. Bloch.

Reply:
removed in https://github.com/apache/maven/pull/827/commits/2b2062338a03246f6308c25429b160a02d1d0f52.